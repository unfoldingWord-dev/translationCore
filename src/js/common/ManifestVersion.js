/**
 * Nested version of the books of the bible object.
 */
export default {
  CURRENT_VERSION: 1
};
