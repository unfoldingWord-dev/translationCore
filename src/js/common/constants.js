import path from "path-extra";
import ospath from "ospath";
import packagefile from '../../../package.json';

export const APP_VERSION = packagefile.version;
export const MIN_COMPATIBLE_VERSION = packagefile.minCompatibleVersion;

export const TRANSLATION_WORDS = 'translationWords';
export const TRANSLATION_NOTES = 'translationNotes';
export const TRANSLATION_ACADEMY = 'translationAcademy';
export const TRANSLATION_HELPS = 'translationHelps';
export const WORD_ALIGNMENT = 'wordAlignment';

export const PROJECTS_PATH = path.join(ospath.home(), 'translationCore', 'projects');
export const USER_RESOURCES_PATH = path.join(ospath.home(), "translationCore", "resources");
export const THELPS_EN_RESOURCES_PATH = path.join(USER_RESOURCES_PATH, 'en', TRANSLATION_HELPS);
export const IMPORTS_PATH = path.join(ospath.home(), 'translationCore', 'imports');
export const PROJECT_INDEX_FOLDER_PATH = path.join('.apps', 'translationCore', 'index');
export const TEMP_IMPORT_PATH = path.join(ospath.home(), 'translationCore', 'imports', 'temp');

export const PROJECT_DOT_APPS_PATH = path.join('.apps', 'translationCore');
export const PROJECT_CHECKDATA_DIRECTORY = path.join(PROJECT_DOT_APPS_PATH, 'checkData');
export const STATIC_RESOURCES_PATH = path.join(__dirname, "../../../tcResources");

export const TC_VERSION = "tc_version";
export const SOURCE_CONTENT_UPDATER_MANIFEST = "source-content-updater-manifest.json";

export const tc_EDIT_VERSION_KEY = "tc_edit_version"; // do not change this string as it will break compatibility with other versions
export const tc_MIN_COMPATIBLE_VERSION_KEY = "tc_min_compatible_version"; // do not change this string as it will break compatibility with other versions
export const tc_MIN_VERSION_ERROR = "TC_MIN_VERSION_ERROR";

export const ORIGINAL_LANGUAGE = "originalLanguage";
export const TARGET_LANGUAGE = "targetLanguage";
export const TARGET_BIBLE = "targetBible";

export const toolCardCategories = {
  'kt': 'Key Terms',
  'names': 'Names',
  'other_terms': 'Other Terms',

  'culture': 'Culture',
  'figures': 'Figures of Speech',
  'numbers': 'Numbers',
  'discourse': 'Discourse',
  'grammar': 'Grammar',
  'other': 'Other'
};
