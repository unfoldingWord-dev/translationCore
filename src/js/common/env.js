// methods for accessing environment variables
// this is needed because many process.env values are no longer defined on the client side (looks to be
// security related).  To get the environment variable on client side can now use remote.process.env.

var isRunningClientSide = !process.env.HOME && !!window; // if environment not defined and we have a window
var processEnv = isRunningClientSide ? window.require('electron').remote.process.env : process.env;

/**
 * get Build number
 * @return {string}
 */
function getBuild() {
  return processEnv.BUILD;
}

/**
 * return appropriate process.env data
 * @return {*}
 */
function getEnv() {
  return processEnv;
}

var env = {
  getBuild,
  getEnv,
};

module.exports = env;
