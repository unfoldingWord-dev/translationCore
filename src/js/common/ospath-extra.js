// this is a patched ospath to work on client side in windows.  The problem was that ospath
// made use of process.env values which are no longer defined on the client side (looks to be
// security related).  To get the environment variable on client side can now use remote.process.env.
// TODO: electronite - investigate better solution
var path = require('path');
var os = require('os');

var processEnv = require('./env').getEnv();

function data () {
  switch (this.__platform || process.platform) {
    case 'win32': return path.resolve(processEnv.APPDATA);
    case 'darwin': return path.resolve(path.join(home.call(this), 'Library/Application Support/'));
    default: return processEnv.XDG_CONFIG_HOME
      ? path.resolve(processEnv.XDG_CONFIG_HOME)
      : path.resolve(path.join(home.call(this), '.config/'));
  }
}

function desktop () {
  return path.join(home.call(this), 'Desktop');
}

function home () {
  // io.js >= 2.3
  if ('homedir' in os) return os.homedir();

  switch (this.__platform || process.platform) {
    case 'win32': return path.resolve(processEnv.USERPROFILE);
    default: return path.resolve(processEnv.HOME)
  }
}

function tmp () {
  switch (this.__platform || process.platform) {
    case 'win32': return path.resolve(processEnv.TEMP);
    default: return path.resolve('/tmp');
  }
}

var ospath = {
  __platform: process.platform,
  data: data,
  desktop: desktop,
  home: home,
  tmp: tmp
};

module.exports = ospath;
