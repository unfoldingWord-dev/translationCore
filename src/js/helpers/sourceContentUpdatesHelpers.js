import fs from 'fs-extra';
import path from 'path-extra';
import ospath from 'ospath';

const USER_RESOURCES_PATH = path.join(ospath.home(), 'translationCore/resources');

const cleanReaddirSync = (path) => {
  const cleanDirectories = fs.readdirSync(path)
    .filter(file => file !== '.DS_Store');

  return cleanDirectories;
};

export const getLocalResourceList = () => {
  const localResourceList = [];
  const resourceLanguages = fs.readdirSync(USER_RESOURCES_PATH)
    .filter(file => path.extname(file) !== '.json' && file !== '.DS_Store');

  for (let i = 0; i < resourceLanguages.length; i++) {
    const languageId = resourceLanguages[i];
    const biblesPath = path.join(USER_RESOURCES_PATH, languageId, 'bibles');
    const tHelpsPath = path.join(USER_RESOURCES_PATH, languageId, 'translationHelps');
    const bibleIds = cleanReaddirSync(biblesPath);
    const tHelpsResources = cleanReaddirSync(tHelpsPath);

    bibleIds.forEach(bibleId => {
      const bibleIdPath = path.join(biblesPath, bibleId);
      const bibleIdContents = cleanReaddirSync(bibleIdPath);

      const resourceManifest = fs.readJsonSync(path.join(bibleIdPath, bibleIdContents[0], 'manifest.json'));
      const localResource = {
        languageId: languageId,
        resourceId: bibleId,
        modifiedTime: resourceManifest.dublin_core.modified
      };

      localResourceList.push(localResource);
    });

    tHelpsResources.forEach(tHelpsId => {
      const tHelpResource = path.join(tHelpsPath, tHelpsId);
      const tHelpResourceContents = cleanReaddirSync(tHelpResource);

      const resourceManifest = fs.readJsonSync(path.join(tHelpResource, tHelpResourceContents[0], 'manifest.json'));
      const localResource = {
        languageId: languageId,
        resourceId: tHelpsId,
        modifiedTime: resourceManifest.dublin_core.modified
      };

      localResourceList.push(localResource);
    });
  }
  return localResourceList;
};
