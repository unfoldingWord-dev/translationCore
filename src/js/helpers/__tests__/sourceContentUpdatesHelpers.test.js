import path from 'path';
import fs from 'fs-extra';
// helpers
import * as sourceContentUpdatesHelpers from '../sourceContentUpdatesHelpers';
// constants
import { USER_RESOURCES_PATH } from '../../common/constants';

describe('migrate resources', () => {
  const sourceResourcesPath = path.join('src', '__tests__', 'fixtures', 'resources');

  beforeEach(() => {
    fs.__resetMockFS();
  });

  it('getResources', () => {
    // given
    const resourcesPath = USER_RESOURCES_PATH;
    const copyResourceFiles = [
      'en',
      'el-x-koine',
      'hbo'];
    fs.__loadFilesIntoMockFs(copyResourceFiles, sourceResourcesPath, resourcesPath);

    // when
    const resources = sourceContentUpdatesHelpers.getLocalResourceList();

    // then
    expect(resources).toMatchSnapshot();
  });
});
