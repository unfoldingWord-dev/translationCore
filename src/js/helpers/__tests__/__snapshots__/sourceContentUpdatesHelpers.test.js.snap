// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migrate resources getResources 1`] = `
Array [
  Object {
    "languageId": "en",
    "modifiedTime": undefined,
    "resourceId": "ult",
  },
  Object {
    "languageId": "en",
    "modifiedTime": undefined,
    "resourceId": "ust",
  },
  Object {
    "languageId": "en",
    "modifiedTime": "2019-05-21T19:48:33+00:00",
    "resourceId": "translationNotes",
  },
  Object {
    "languageId": "en",
    "modifiedTime": "2019-05-17T19:25:28+00:00",
    "resourceId": "translationWords",
  },
  Object {
    "languageId": "el-x-koine",
    "modifiedTime": undefined,
    "resourceId": "ugnt",
  },
  Object {
    "languageId": "hbo",
    "modifiedTime": undefined,
    "resourceId": "uhb",
  },
]
`;
